### PR TITLE
Fix parsing shape

### LINKED ISSUE #559: Shape is not parsed correctly
**Description**
When parsing `NUMERIC(name, [5,10])` in
```SQL
SELECT ...
TRAIN ...
COLUMN NUMERIC(name, [5,10])
...
```
`[5,10]` won't be parsed as `[`, `5`, `,`, `10`, and `]`, but `[`, `5,10`, `]` instead.

**Reproduction Steps**
```go
func TestNumericColumn(t *testing.T) {
	a := assert.New(t)
	parser := newParser()

	// Note it is [5,10] without the space
	normal := statementWithColumn("NUMERIC(c2, [5,10])")
	moreArgs := statementWithColumn("NUMERIC(c1, 100, args)")
	badShape := statementWithColumn("NUMERIC(c1, bad)")

	r, e := parser.Parse(normal)
	a.NoError(e)
	c := r.columns["feature_columns"]
	fcs, _, e := resolveTrainColumns(&c)
	a.NoError(e)
	nc, ok := fcs[0].(*numericColumn)
	a.True(ok)
	code, e := nc.GenerateCode()
	a.NoError(e)
	a.Equal("c2", nc.Key)
	a.Equal([]int{5, 10}, nc.Shape)
	a.Equal("tf.feature_column.numeric_column(\"c2\", shape=[5,10])", code)
}
```

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #559.

### CODE REVIEW

File: sql/lexer.go [hunk 1]
```diff
@@ -138,8 +138,7 @@ func (l *lexer) emitIdentOrKeyword(lval *sqlSymType) int {
 }
 
 var (
-	// Stolen https://www.regular-expressions.info/floatingpoint.html.
-	reNumber = regexp.MustCompile("[-+]?[0-9]*.?[0-9]+([eE][-+]?[0-9]+)?")
+	reNumber = regexp.MustCompile("[-+]?[0-9]*[.]?[0-9]+([eE][-+]?[0-9]+)?")
```
Comment (tonyyang-svail) [inline sql/lexer.go:141]: `.` will match any character. What we want is the exact match of character `.`.

Comment (typhoonzero) [reply sql/lexer.go:141]: does that mean should `\.` also works?

**Review by typhoonzero — APPROVED** [review]
LGTM

**Comment by tonyyang-svail** [comment]
> does that mean should `\.` also works?

@typhoonzero Right. Both `"\\."` and `"[.]"` will work.

**Pull request merged**